node/pkg/allocateip/allocateip_test.go

```diff
@@ -56,6 +56,14 @@ func setTunnelAddressForNode(tunnelType string, n *libapi.Node, addr string) {
 		} else {
 			n.Spec.Wireguard = nil
 		}
+	} else if tunnelType == ipam.AttributeTypeWireguardV6 {
```

Comment:
Same for the else branch - we only want to set it to `nil` if there isn't another field in the Wireguard section, otherwise this function will be removing fields that the caller doesn't expect it to.

Reply:
You're right, thank you for catching this! I'll fix those...

Reply:
Makes me think we probably are missing some dual-stack coverage in the tests, if this isn't causing a test failure somewhere.

Reply:
I added dual stack tests (for VXLAN as well), but the ones I did still didn't use that specific function... Let me know your thoughts